src/main/java/htsjdk/samtools/util/CigarUtil.java

```diff
@@ -140,6 +157,16 @@ public static void softClip3PrimeEndOfRead(SAMRecord rec, final int clipFrom) {
         }
         rec.setCigar(newCigar);
 
+        // If hard-clipping, remove the hard-clipped bases from the read
+        if(clippingOperator == CigarOperator.HARD_CLIP) {
+            byte[] bases = rec.getReadBases();
+            if(rec.getReadNegativeStrandFlag()) {
+                rec.setReadBases(Arrays.copyOfRange(bases, clipFrom, bases.length));
```

Comment:
I think this actually needs to be `rec.setReadBases(Arrays.copyOfRange(bases, bases.length - clipFrom, bases.length));` . 
Also should add some tests to make sure the changes to the read bases are done correctly, as this is ripe for nasty off-by-1 errors and such things.

Reply:
looks like @lbergelson already requested these tests.

Reply:
Nice catch @kachulis.